### ISSUE TITLE
Display snapshot metadata required for recovery

### ISSUE THREAD

**Issue by mikewallace1979** [EnterpriseDB/barman#696]
This needs to be supported by both barman and barman-cloud.

Not sure whether to make this a new command, add the metadata to show-backup, or to have `barman recover` (and `barman-cloud-restore`) do it as a default - i.e. "This is a snapshot backup. You need to make sure I'm pointing to a node with the following disks: ...." and allow the recovery preparation to proceed by prompt or a force option.

**Comment by mikewallace1979** [comment]
Probably just add the metadata to show-backup, something like:

```
"snapshot_metadata": {
    "gce_project": "<GCE_PROJECT_ID>",
    "snapshots": [{
            "snapshot_id": "<SNAPSHOT_ID>"
            "device_name": "<DEVICE_NAME>",
            "mount_point": "<MOUNT_POINT>"
        }
    ]
}
```

**Comment by mikewallace1979** [comment]
Closed by #728
